### PR TITLE
#43 version name is unspecified

### PR DESCRIPTION
**Pull request by ouchadam**
Deprecates the "version" param is favour of "publishVersion" to avoid conflicts with gradle's `project.version` when using `properties`

Fixes unspecified version name by applying the same property finding logic to both the `maven publication` and the `bintray configuration`

### CODE REVIEW

**Comment by ouchadam** [comment]
fixes #43

File: core/src/main/groovy/com/novoda/gradle/release/PropertyFinder.groovy [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    def getBintrayKey() {
+        getString(project, 'bintrayKey', extension.bintrayKey)
+    }
+
+    def getDryRun() {
+        getBoolean(project, 'dryRun', extension.dryRun)
+    }
+
+    def getPublishVersion() {
+        getString(project, 'publishVersion', extension.publishVersion == null ? extension.version : extension.publishVersion)
```
Comment (blundell) [inline core/src/main/groovy/com/novoda/gradle/release/PropertyFinder.groovy]: `getString(project, 'publishVersion', extension.publishVersion ?: extension.version)`

http://docs.groovy-lang.org/latest/html/documentation/index.html#_elvis_operator

The "Elvis operator" is a shortening of the ternary operator. Basically means if not null null return myself

> with the Elvis operator, the value which is tested is used if it is not false or null

File: core/src/main/groovy/com/novoda/gradle/release/PublishExtension.groovy [hunk 2]
```diff
@@ -7,7 +7,11 @@ class PublishExtension {
 
     String groupId
     String artifactId
+
+    @Deprecated
```
Comment (blundell) [inline core/src/main/groovy/com/novoda/gradle/release/PublishExtension.groovy]:
```
/**
* Please use #publishVersion and see https://github.com/novoda/bintray-release/issues/43 for why
**/
@Deprecated
String version
```

**Comment by blundell** [comment]
nice action :+1:

File: core/src/main/groovy/com/novoda/gradle/release/PublishExtension.groovy (hunk 2 above)
Comment (ouchadam) [inline core/src/main/groovy/com/novoda/gradle/release/PublishExtension.groovy]: :+1: definitely

File: core/src/main/groovy/com/novoda/gradle/release/PropertyFinder.groovy (hunk 1 above)
Comment (ouchadam) [inline core/src/main/groovy/com/novoda/gradle/release/PropertyFinder.groovy]: that is such a good idea :+1:

**Comment by ouchadam** [comment]
@blundell comments addressed

**Comment by blundell** [comment]
:+1:

**Comment by xrigau** [comment]
:+1:

**Pull request merged**